src/oc/hocedit.cpp

```diff
@@ -64,17 +64,11 @@ hocedit.c,v
 #include <hocparse.h>
 #include <setjmp.h>
 
-
-#if MAC
-#define OCSMALL 1
-#endif
-
-#if defined(WITHOUT_MEMACS)
-#undef OCSMALL
+#ifndef WITHOUT_MEMACS
 #define OCSMALL 1
 #endif
```

Comment:
this is different than original code above. I assume you have checked if it's problematic.

Reply:
Yes (same as above, underlying undefined symbols)

Reply:
Side note: ` * OCSMALL used to throw out things not needed by teaching programs`
I believe most of the logic was not followed through all the way (i.e MAC or LINUX)

Reply:
refactored some more all `hocedit`